Build failure: omnix

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

Build on the latest nixos-unstable branch

```shell
$ nix build "github:nixos/nixpkgs?ref=cad22e7d996aea55ecab064e84834289143e44a0#omnix"
```

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/job/nixos/unstable/nixpkgs.omnix.x86_64-linux

### Relevant log output

```console
Using versionCheckHook
Running phase: unpackPhase
unpacking source archive /nix/store/n74mm3shvbnzr2p2ml6hfz7sg0ab0jvd-source
source root is source
Executing cargoSetupPostUnpackHook
Finished cargoSetupPostUnpackHook
Running phase: patchPhase
Executing cargoSetupPostPatchHook
Validating consistency between /build/source/Cargo.lock and /build/omnix-1.3.0-vendor/Cargo.lock
Finished cargoSetupPostPatchHook
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
Running phase: buildPhase
Executing cargoBuildHook
cargoBuildHook flags: -j 16 --target x86_64-unknown-linux-gnu --offline --profile release
   Compiling proc-macro2 v1.0.79
   Compiling unicode-ident v1.0.12
   Compiling cfg-if v1.0.0
   Compiling serde v1.0.204
   Compiling libc v0.2.159
   Compiling log v0.4.22
   Compiling autocfg v1.1.0
   Compiling memchr v2.6.4
   Compiling smallvec v1.13.2
   Compiling once_cell v1.18.0
   Compiling crossbeam-utils v0.8.20
   Compiling pin-project-lite v0.2.13
   Compiling stable_deref_trait v1.2.0
   Compiling version_check v0.9.4
   Compiling bitflags v2.4.1
   Compiling scopeguard v1.2.0
   Compiling thiserror v1.0.63
   Compiling litemap v0.7.4
   Compiling writeable v0.5.5
   Compiling itoa v1.0.9
   Compiling pkg-config v0.3.27
   Compiling icu_locid_transform_data v1.5.0
   Compiling semver v1.0.22
   Compiling parking_lot_core v0.9.9
   Compiling lock_api v0.4.11
   Compiling icu_properties_data v1.5.0
   Compiling simd-adler32 v0.3.7
   Compiling tracing-core v0.1.32
   Compiling hashbrown v0.14.3
   Compiling write16 v1.0.0
   Compiling either v1.9.0
   Compiling utf8_iter v1.0.4
   Compiling equivalent v1.0.1
   Compiling futures-core v0.3.31
   Compiling utf16_iter v1.0.5
   Compiling icu_normalizer_data v1.5.0
   Compiling num-traits v0.2.19
   Compiling regex-syntax v0.8.2
   Compiling rayon-core v1.12.0
   Compiling aho-corasick v1.1.2
   Compiling anstyle v1.0.8
   Compiling adler2 v2.0.0
   Compiling quote v1.0.35
   Compiling crossbeam-epoch v0.9.18
   Compiling fnv v1.0.7
   Compiling futures-io v0.3.31
   Compiling syn v1.0.109
   Compiling percent-encoding v2.3.1
   Compiling bytes v1.7.2
   Compiling syn v2.0.53
   Compiling bytemuck v1.19.0
   Compiling form_urlencoded v1.2.1
   Compiling crossbeam-deque v0.8.5
   Compiling miniz_oxide v0.8.0
   Compiling indexmap v2.1.0
   Compiling crc32fast v1.4.2
   Compiling cc v1.0.83
   Compiling signal-hook-registry v1.4.1
   Compiling mio v0.8.11
   Compiling ryu v1.0.15
   Compiling parking_lot v0.12.1
   Compiling libm v0.2.8
   Compiling bitflags v1.3.2
   Compiling utf8parse v0.2.1
   Compiling rustix v0.38.37
   Compiling anstyle-parse v0.2.2
   Compiling flate2 v1.0.34
   Compiling num_cpus v1.16.0
   Compiling socket2 v0.5.5
   Compiling fastrand v2.0.1
   Compiling linux-raw-sys v0.4.14
   Compiling serde_json v1.0.120
   Compiling anstyle-query v1.1.1
   Compiling is_terminal_polyfill v1.70.0
   Compiling colorchoice v1.0.0
   Compiling parking v2.2.0
   Compiling anstream v0.6.14
   Compiling num-integer v0.1.46
   Compiling rayon v1.8.0
   Compiling concurrent-queue v2.3.0
   Compiling anyhow v1.0.89
   Compiling regex-automata v0.4.3
   Compiling powerfmt v0.2.0
   Compiling num-bigint v0.4.6
   Compiling event-listener v4.0.0
   Compiling slotmap v1.0.7
   Compiling deranged v0.3.9
   Compiling core_maths v0.1.0
   Compiling clap_lex v0.7.4
   Compiling unicode-width v0.1.13
   Compiling bit-vec v0.6.3
   Compiling ident_case v1.0.1
   Compiling float-cmp v0.9.0
   Compiling vcpkg v0.2.15
   Compiling strsim v0.11.1
   Compiling minimal-lexical v0.2.1
   Compiling heck v0.5.0
   Compiling aligned-vec v0.5.0
   Compiling num-conv v0.1.0
   Compiling adler v1.0.2
   Compiling same-file v1.0.6
   Compiling time-core v0.1.2
   Compiling paste v1.0.15
   Compiling strsim v0.10.0
   Compiling time v0.3.36
   Compiling walkdir v2.4.0
   Compiling openssl-sys v0.9.107
   Compiling miniz_oxide v0.7.1
   Compiling v_frame v0.3.8
   Compiling num-rational v0.4.2
   Compiling nom v7.1.3
   Compiling clap_builder v4.5.27
   Compiling strict-num v0.1.1
   Compiling bit-set v0.5.3
   Compiling event-listener-strategy v0.4.0
   Compiling ttf-parser v0.24.1
   Compiling synstructure v0.13.1
   Compiling darling_core v0.20.3
   Compiling regex v1.10.2
   Compiling onig_sys v69.8.1
   Compiling quick-xml v0.32.0
   Compiling fdeflate v0.3.5
   Compiling lazy_static v1.4.0
   Compiling base64 v0.22.1
   Compiling rust_hawktracer_proc_macro v0.4.1
   Compiling arrayref v0.3.9
   Compiling rust_hawktracer_normal_macro v0.4.1
   Compiling roxmltree v0.20.0
   Compiling tinyvec_macros v0.1.1
   Compiling tiny-skia-path v0.11.4
   Compiling tinyvec v1.6.0
   Compiling rust_hawktracer v0.7.0
   Compiling png v0.17.14
   Compiling fontconfig-parser v0.5.7
   Compiling serde_derive v1.0.204
   Compiling zerofrom-derive v0.1.5
   Compiling yoke-derive v0.7.5
   Compiling zerovec-derive v0.10.3
   Compiling displaydoc v0.2.5
   Compiling icu_provider_macros v1.5.0
   Compiling thiserror-impl v1.0.63
   Compiling tracing-attributes v0.1.27
   Compiling tokio-macros v2.3.0
   Compiling clap_derive v4.5.24
   Compiling darling_macro v0.20.3
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_st_init_strend_table_with_size':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:588:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *, st_str_end_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   588 |     str_end_cmp,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:588:5: note: (near initialization for 'hashType.compare')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:550:1: note: 'str_end_cmp' declared here
warning: onig_sys@69.8.1:   550 | str_end_cmp(st_str_end_key* x, st_str_end_key* y)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:589:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   589 |     str_end_hash,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:589:5: note: (near initialization for 'hashType.hash')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:571:1: note: 'str_end_hash' declared here
warning: onig_sys@69.8.1:   571 | str_end_hash(st_str_end_key* x)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_st_init_callout_name_table_with_size':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:678:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *, st_callout_name_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   678 |     callout_name_table_cmp,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:678:5: note: (near initialization for 'hashType.compare')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:637:1: note: 'callout_name_table_cmp' declared here
warning: onig_sys@69.8.1:   637 | callout_name_table_cmp(st_callout_name_key* x, st_callout_name_key* y)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:679:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *)' [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   679 |     callout_name_table_hash,
warning: onig_sys@69.8.1:       |     ^~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:679:5: note: (near initialization for 'hashType.hash')
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:660:1: note: 'callout_name_table_hash' declared here
warning: onig_sys@69.8.1:   660 | callout_name_table_hash(st_callout_name_key* x)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'names_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:804:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   804 |     onig_st_foreach(t, i_free_name_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}
warning: onig_sys@69.8.1: In file included from oniguruma/src/regparse.c:37:
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:789:1: note: 'i_free_name_entry' declared here
warning: onig_sys@69.8.1:   789 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_foreach_name':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:873:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   873 |     onig_st_foreach(t, i_names, (HashDataType )&narg);
warning: onig_sys@69.8.1:       |                        ^~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:846:1: note: 'i_names' declared here
warning: onig_sys@69.8.1:   846 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)
warning: onig_sys@69.8.1:       | ^~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'onig_renumber_name_table':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:901:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:   901 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:879:1: note: 'i_renumber_name' declared here
warning: onig_sys@69.8.1:   879 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'callout_name_table_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1386:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1386 |     onig_st_foreach(t, i_free_callout_name_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1370:1: note: 'i_free_callout_name_entry' declared here
warning: onig_sys@69.8.1:  1370 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1884:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1884 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,
warning: onig_sys@69.8.1:       |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                                                        |
warning: onig_sys@69.8.1:       |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1866:1: note: 'i_callout_callout_list_set' declared here
warning: onig_sys@69.8.1:  1866 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c: In function 'callout_tag_table_clear':
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1932:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]
warning: onig_sys@69.8.1:  1932 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);
warning: onig_sys@69.8.1:       |                        ^~~~~~~~~~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1:       |                        |
warning: onig_sys@69.8.1:       |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}
warning: onig_sys@69.8.1: oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}
warning: onig_sys@69.8.1:    55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));
warning: onig_sys@69.8.1:       |                               ^~~~~~~~~~~~~~~~
warning: onig_sys@69.8.1: oniguruma/src/st.h:35:18: note: in definition of macro '_'
warning: onig_sys@69.8.1:    35 | # define _(args) args
warning: onig_sys@69.8.1:       |                  ^~~~
warning: onig_sys@69.8.1: oniguruma/src/regparse.c:1922:1: note: 'i_free_callout_tag_entry' declared here
warning: onig_sys@69.8.1:  1922 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)
warning: onig_sys@69.8.1:       | ^~~~~~~~~~~~~~~~~~~~~~~~
error: failed to run custom build command for `onig_sys v69.8.1`

Caused by:
  process didn't exit successfully: `/build/source/target/release/build/onig_sys-6884ef217d71f459/build-script-build` (exit status: 1)
  --- stdout
  cargo:rerun-if-env-changed=RUSTONIG_DYNAMIC_LIBONIG
  cargo:rerun-if-env-changed=RUSTONIG_STATIC_LIBONIG
  cargo:rerun-if-env-changed=RUSTONIG_SYSTEM_LIBONIG
  TARGET = Some("x86_64-unknown-linux-gnu")
  OPT_LEVEL = Some("z")
  HOST = Some("x86_64-unknown-linux-gnu")
  cargo:rerun-if-env-changed=CC_x86_64-unknown-linux-gnu
  CC_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CC_x86_64_unknown_linux_gnu
  CC_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = Some("/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc")
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some("false")
  CARGO_CFG_TARGET_FEATURE = Some("fxsr,sse,sse2")
  cargo:rerun-if-env-changed=CFLAGS_x86_64-unknown-linux-gnu
  CFLAGS_x86_64-unknown-linux-gnu = None
  cargo:rerun-if-env-changed=CFLAGS_x86_64_unknown_linux_gnu
  CFLAGS_x86_64_unknown_linux_gnu = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-Os" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out/oniguruma/src/regexec.o" "-c" "oniguruma/src/regexec.c"
  exit status: 0
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-Os" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out/oniguruma/src/regerror.o" "-c" "oniguruma/src/regerror.c"
  exit status: 0
  running: "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-Os" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out/oniguruma/src/regparse.o" "-c" "oniguruma/src/regparse.c"
  cargo:warning=oniguruma/src/regparse.c: In function 'onig_st_init_strend_table_with_size':

  cargo:warning=oniguruma/src/regparse.c:588:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *, st_str_end_key *)' [-Wincompatible-pointer-types]

  cargo:warning=  588 |     str_end_cmp,

  cargo:warning=      |     ^~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:588:5: note: (near initialization for 'hashType.compare')

  cargo:warning=oniguruma/src/regparse.c:550:1: note: 'str_end_cmp' declared here

  cargo:warning=  550 | str_end_cmp(st_str_end_key* x, st_str_end_key* y)

  cargo:warning=      | ^~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:589:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_str_end_key *)' [-Wincompatible-pointer-types]

  cargo:warning=  589 |     str_end_hash,

  cargo:warning=      |     ^~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:589:5: note: (near initialization for 'hashType.hash')

  cargo:warning=oniguruma/src/regparse.c:571:1: note: 'str_end_hash' declared here

  cargo:warning=  571 | str_end_hash(st_str_end_key* x)

  cargo:warning=      | ^~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'onig_st_init_callout_name_table_with_size':

  cargo:warning=oniguruma/src/regparse.c:678:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *, st_callout_name_key *)' [-Wincompatible-pointer-types]

  cargo:warning=  678 |     callout_name_table_cmp,

  cargo:warning=      |     ^~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:678:5: note: (near initialization for 'hashType.compare')

  cargo:warning=oniguruma/src/regparse.c:637:1: note: 'callout_name_table_cmp' declared here

  cargo:warning=  637 | callout_name_table_cmp(st_callout_name_key* x, st_callout_name_key* y)

  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:679:5: error: initialization of 'int (*)(void)' from incompatible pointer type 'int (*)(st_callout_name_key *)' [-Wincompatible-pointer-types]

  cargo:warning=  679 |     callout_name_table_hash,

  cargo:warning=      |     ^~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c:679:5: note: (near initialization for 'hashType.hash')

  cargo:warning=oniguruma/src/regparse.c:660:1: note: 'callout_name_table_hash' declared here

  cargo:warning=  660 | callout_name_table_hash(st_callout_name_key* x)

  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'names_clear':

  cargo:warning=oniguruma/src/regparse.c:804:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning=  804 |     onig_st_foreach(t, i_free_name_entry, 0);

  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~

  cargo:warning=      |                        |

  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, void *) {aka int (*)(unsigned char *, NameEntry *, void *)}

  cargo:warning=In file included from oniguruma/src/regparse.c:37:

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, void *)' {aka 'int (*)(unsigned char *, NameEntry *, void *)'}

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:789:1: note: 'i_free_name_entry' declared here

  cargo:warning=  789 | i_free_name_entry(UChar* key, NameEntry* e, void* arg ARG_UNUSED)

  cargo:warning=      | ^~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'onig_foreach_name':

  cargo:warning=oniguruma/src/regparse.c:873:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning=  873 |     onig_st_foreach(t, i_names, (HashDataType )&narg);

  cargo:warning=      |                        ^~~~~~~

  cargo:warning=      |                        |

  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, INamesArg *) {aka int (*)(unsigned char *, NameEntry *, INamesArg *)}

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, INamesArg *)' {aka 'int (*)(unsigned char *, NameEntry *, INamesArg *)'}

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:846:1: note: 'i_names' declared here

  cargo:warning=  846 | i_names(UChar* key ARG_UNUSED, NameEntry* e, INamesArg* arg)

  cargo:warning=      | ^~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'onig_renumber_name_table':

  cargo:warning=oniguruma/src/regparse.c:901:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning=  901 |     onig_st_foreach(t, i_renumber_name, (HashDataType )map);

  cargo:warning=      |                        ^~~~~~~~~~~~~~~

  cargo:warning=      |                        |

  cargo:warning=      |                        int (*)(OnigUChar *, NameEntry *, GroupNumMap *) {aka int (*)(unsigned char *, NameEntry *, GroupNumMap *)}

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, NameEntry *, GroupNumMap *)' {aka 'int (*)(unsigned char *, NameEntry *, GroupNumMap *)'}

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:879:1: note: 'i_renumber_name' declared here

  cargo:warning=  879 | i_renumber_name(UChar* key ARG_UNUSED, NameEntry* e, GroupNumMap* map)

  cargo:warning=      | ^~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'callout_name_table_clear':

  cargo:warning=oniguruma/src/regparse.c:1386:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning= 1386 |     onig_st_foreach(t, i_free_callout_name_entry, 0);

  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=      |                        |

  cargo:warning=      |                        int (*)(st_callout_name_key *, CalloutNameEntry *, void *)

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(st_callout_name_key *, CalloutNameEntry *, void *)'

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:1370:1: note: 'i_free_callout_name_entry' declared here

  cargo:warning= 1370 | i_free_callout_name_entry(st_callout_name_key* key, CalloutNameEntry* e,

  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'setup_ext_callout_list_values':

  cargo:warning=oniguruma/src/regparse.c:1884:56: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning= 1884 |     onig_st_foreach((CalloutTagTable *)ext->tag_table, i_callout_callout_list_set,

  cargo:warning=      |                                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=      |                                                        |

  cargo:warning=      |                                                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:1866:1: note: 'i_callout_callout_list_set' declared here

  cargo:warning= 1866 | i_callout_callout_list_set(UChar* key, CalloutTagVal e, void* arg)

  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/regparse.c: In function 'callout_tag_table_clear':

  cargo:warning=oniguruma/src/regparse.c:1932:24: error: passing argument 2 of 'onig_st_foreach' from incompatible pointer type [-Wincompatible-pointer-types]

  cargo:warning= 1932 |     onig_st_foreach(t, i_free_callout_tag_entry, 0);

  cargo:warning=      |                        ^~~~~~~~~~~~~~~~~~~~~~~~

  cargo:warning=      |                        |

  cargo:warning=      |                        int (*)(OnigUChar *, CalloutTagVal,  void *) {aka int (*)(unsigned char *, long int,  void *)}

  cargo:warning=oniguruma/src/st.h:55:31: note: expected 'int (*)(void)' but argument is of type 'int (*)(OnigUChar *, CalloutTagVal,  void *)' {aka 'int (*)(unsigned char *, long int,  void *)'}

  cargo:warning=   55 | int st_foreach _((st_table *, int (*)(ANYARGS), st_data_t));

  cargo:warning=      |                               ^~~~~~~~~~~~~~~~

  cargo:warning=oniguruma/src/st.h:35:18: note: in definition of macro '_'

  cargo:warning=   35 | # define _(args) args

  cargo:warning=      |                  ^~~~

  cargo:warning=oniguruma/src/regparse.c:1922:1: note: 'i_free_callout_tag_entry' declared here

  cargo:warning= 1922 | i_free_callout_tag_entry(UChar* key, CalloutTagVal e, void* arg ARG_UNUSED)

  cargo:warning=      | ^~~~~~~~~~~~~~~~~~~~~~~~

  exit status: 1

  --- stderr


  error occurred: Command "/nix/store/r9wbjib6xxjkyb9yvjvrkl4sq61i2lyn-gcc-wrapper-15.2.0/bin/cc" "-Os" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m64" "-I" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out" "-I" "oniguruma/src" "-DHAVE_UNISTD_H=1" "-DHAVE_SYS_TYPES_H=1" "-DHAVE_SYS_TIME_H=1" "-o" "/build/source/target/x86_64-unknown-linux-gnu/release/build/onig_sys-02b90454dcd6bc2f/out/oniguruma/src/regparse.o" "-c" "oniguruma/src/regparse.c" with args "cc" did not execute successfully (status code exit status: 1).


warning: build failed, waiting for other jobs to finish...
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 26.05 (Yarara), 26.05.20251228.c0b0e0f`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.32.4`
 - nixpkgs: `/nix/store/xzl5dhddgjxp2p18h5qpkhvlkp8128j1-source`

### Notify maintainers

@shivaraj-bh 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/om/omnix/package.nix
+++ b/pkgs/by-name/om/omnix/package.nix
@@ -11,16 +11,16 @@
 }:
 rustPlatform.buildRustPackage (finalAttrs: {
   pname = "omnix";
-  version = "1.3.0";
+  version = "1.3.2";
 
   src = fetchFromGitHub {
     owner = "juspay";
     repo = "omnix";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-wxBZLeYoYIQlZLsWpJty95fUmEbgHPCwbn392b9n3NY=";
+    hash = "sha256-D9rAVsSFooVWpSX//gTcRcmgiAjwZYNRMDIctMmwnho=";
   };
 
-  cargoHash = "sha256-qZv/4P9f87Glif2VsdrY23JGw6bSSQYdekF6erY5owk=";
+  cargoHash = "sha256-3zWbhuZzqkxgM0Js3luR6+Yr5/UGn1EoL6OqxPt94JM=";
 
   nativeBuildInputs = [
     installShellFiles

